packages/web3-eth-accounts/src/index.js

```diff
@@ -201,30 +201,32 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
                     delete transaction.hardfork;
                 }
             }
-
+            
             if (privateKey.startsWith('0x')) {
                 privateKey = privateKey.substring(2);
             }
-
-            var ethTx = new Transaction(transaction, transactionOptions);
-
-            ethTx.sign(Buffer.from(privateKey, 'hex'));
-
-            var validationResult = ethTx.validate(true);
-
-            if (validationResult !== '') {
-                throw new Error('Signer Error: ' + validationResult);
+            
+            var ethTx = Transaction.fromTxData(transaction, transactionOptions);
+            var signedTx = ethTx.sign(Buffer.from(privateKey, 'hex'));
+            var validationResult = signedTx.validate(true);
+
+            if (!validationResult || validationResult.length > 0) {
```

Comment:
Since we're using `validate(true)` it will always return the array of errors `[]string` so I think we want to do:
```
if (validationResult.length > 0) { ... }
```